<a href="https://colab.research.google.com/github/nueveonce/DS-Etapa-2-final/blob/main/Proyecto_Final_Toledo_Roberto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Desarrollo de un modelo de Machine Learning para predecir el riesgo de infarto.

# ▶️ Abstract

 Este conjunto de datos proporciona una amplia gama de características relevantes para la salud cardíaca y las opciones de estilo de vida, que abarcan detalles específicos del paciente, como edad, sexo, niveles de colesterol, presión arterial, frecuencia cardíaca e indicadores como diabetes, antecedentes familiares, hábito de fumar, obesidad y consumo de alcohol. Además, se incluyen factores de estilo de vida como horas de ejercicio, hábitos alimentarios, niveles de estrés y horas de sedentarismo. Se consideran aspectos médicos que comprenden problemas cardíacos previos, uso de medicamentos y niveles de triglicéridos. Se incorporan aspectos socioeconómicos como ingresos y atributos geográficos como país, continente y hemisferio. El conjunto de datos, que consta de 8763 registros de pacientes de todo el mundo, culmina en una característica de clasificación binaria que denota la presencia o ausencia de un riesgo de ataque cardíaco, lo que proporciona un recurso integral para el análisis predictivo y la investigación en salud cardiovascular.

# ▶️ Contexto empresarial.

Fundación Cruz Roja Argentina es una organización líder en innovación tecnológica para la salud preventiva, especializada en desarrollar soluciones de inteligencia artificial para la detección temprana de riesgos cardiovasculares. La compañía busca revolucionar la medicina preventiva mediante el uso de análisis de datos predictivos que permitan identificar potenciales riesgos de ataques cardíacos antes de que ocurran.



# ▶️ Problema comercial

El mercado actual enfrenta desafíos críticos en la prevención de enfermedades cardiovasculares:
- Las enfermedades cardíacas representan la principal causa de muerte a nivel mundial
- Existe una alta tasa de mortalidad por diagnósticos tardíos
- Los costos de tratamiento post-ataque cardíaco son significativamente elevados
- Hay una necesidad urgente de herramientas predictivas precisas y accesibles

La empresa necesita desarrollar un modelo predictivo que:
- Identifique pacientes con alto riesgo de ataque cardíaco
- Proporcione alertas tempranas para intervención médica
- Reduzca costos de tratamiento
- Mejore la calidad de vida de los pacientes




# ▶️ Contexto analítico

Construir un modelo de machine learning que prediga con alta precisión el riesgo de ataque cardíaco basado en múltiples variables de salud, estilo de vida y factores socioeconómicos.

# ▶️ Acerca del Dataset

## ⏩ Descripción del conjunto de columnas del dataset

+ **Identificación del paciente**: identificador único para cada paciente
+ **Edad** - Edad del paciente
+ **Sexo** - Género del paciente (Masculino/Femenino)
+ **Colesterol** - El colesterol LDL, también conocido como colesterol "malo", se acumula en las paredes de las arterias y puede contribuir a la obstrucción de estas.

<table style="border-collapse: collapse; width: 70%; margin: 20px 0; font-size:15px">
  <thead>
    <tr style="background-color: #16180c; color: white;">
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Nivel de Colesterol LDL (mg/dL)</th>
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Categoría</th>
    </tr>
  </thead>
  <tbody>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Menos de 100</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Óptimo</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">100-129</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Casi óptimo o poco mejor que óptimo</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">130-159</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Límite alto</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">160-189</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Alto</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">190 o mayor</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Muy alto</td>
    </tr>
  </tbody>
</table>

+ **Presión arterial** : presión arterial del paciente (sistólica/diastólica). La presión arterial sistólica es la presión en las arterias cuando el corazón late, y la diastólica es la presión entre un latido y el siguiente.

<table style="border-collapse: collapse; width: 70%; margin: 20px 0;font-size:15px">
  <thead>
    <tr style="background-color: #16180c; color: white;">
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Categoría</th>
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Presión Arterial Sistólica (mmHg)</th>
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Presión Arterial Diastólica (mmHg)</th>
    </tr>
  </thead>
  <tbody>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Normal</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Menor a 120</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Menor a 80</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Elevada</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">120-129</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Menor a 80</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Hipertensión grado 1</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">130-139</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">80-89</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Hipertensión grado 2</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">140 o mayor</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">90 o mayor</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Crisis hipertensiva</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Mayor a 180</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Mayor a 120</td>
    </tr>
  </tbody>
</table>

+ **Frecuencia cardíaca** - La frecuencia cardíaca es el número de veces que el corazón late en un minuto, también conocida como pulso. En reposo, una frecuencia cardíaca normal oscila entre 60 y 100 latidos por minuto.
+ **Diabetes** - Si el paciente tiene diabetes (Sí/No). La diabetes es una enfermedad crónica que se caracteriza por niveles elevados de azúcar en la sangre. Esto se debe a que el cuerpo no produce suficiente insulina, la hormona que regula el azúcar en la sangre, o no la usa de forma adecuada
+ **Antecedentes familiares** : antecedentes familiares de problemas relacionados con el corazón (1: Sí, 0: No)
+ **Tabaquismo** - Estado de tabaquismo del paciente (1: Fumador, 0: No fumador)
+ **Obesidad** - Estado de obesidad del paciente (1: Obeso, 0: No obeso)
+ **Consumo de alcohol** : nivel de consumo de alcohol por parte del paciente (Ninguno/Leve/Moderado/Alto)
+ **Horas de ejercicio por semana** - Número de horas de ejercicio por semana
+ **Dieta** - Hábitos alimentarios del paciente (Saludable/Regular/No saludable)
+ **Problemas cardíacos previos** : Problemas cardíacos previos del paciente (1: Sí, 0: No)
+ **Uso de medicamentos** - Uso de medicamentos por parte del paciente (1: Sí, 0: No)
+ **Nivel de estrés** : nivel de estrés informado por el paciente (1-10)
+ **Horas de sedentarismo por día** - Horas de actividad sedentaria por día
+ **Ingresos** - Nivel de ingresos del paciente
+ **IMC** - Índice de Masa Corporal (IMC) del paciente
+ **Triglicéridos** - Niveles de triglicéridos del paciente

<table style="border-collapse: collapse; width: 70%; margin: 20px 0;font-size:15px">
  <thead>
    <tr style="background-color: #16180c; color: white;">
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Categoría</th>
      <th style="padding: 12px; border: 2px solid #2b3018; text-align: left;">Niveles de Triglicéridos (mg/dl)</th>
    </tr>
  </thead>
  <tbody>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Saludables (adultos)</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Menos de 150</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Ligeramente altos</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">150-199</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Altos</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">200-499</td>
    </tr>
    <tr style="background-color: #414823;">
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">Muy altos</td>
      <td style="padding: 12px; border: 2px solid #2b3018; color: white;">500 o más</td>
    </tr>
  </tbody>
</table>


+ **Días de actividad física por semana** - Días de actividad física por semana
+ **Horas de sueño por día** - Horas de sueño por día
+ **País** - País del paciente
+ **Continente** - Continente donde reside el paciente
+ **Hemisferio** - Hemisferio donde reside el paciente
+ **Riesgo de ataque cardíaco** : presencia de riesgo de ataque cardíaco (1: Sí, 0: No)

# ▶️ Importación de librerias y carga de dataset

In [50]:
import pandas as pd
import time
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
from google.colab import drive
drive.mount('/content/drive')



try:
    print("Intentando cargar el DataFrame...")
    start_time = time.time()

    ruta = '/content/drive/MyDrive/Colab'+' '+'Notebooks/Datasets/heart_attack_prediction_dataset.csv'
    df = pd.read_csv(ruta, sep=";")

    print(f"Carga exitosa! Tiempo: {time.time()-start_time:.2f} segundos")

except FileNotFoundError:
    print("ERROR: Archivo no encontrado. Verifica la ruta.")
    print(ruta)
except Exception as e:
    print(f"ERROR inesperado: {str(e)}")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Intentando cargar el DataFrame...
Carga exitosa! Tiempo: 0.04 segundos


# ▶️ EDA -  Exploración de datos del dataset


In [51]:
# Visualización de las primeros 5 filas y últimas 5 filas del dataframe
df.head(10)

Patient ID   Age     Sex  Cholesterol Blood Pressure  Heart Rate  Diabetes  \
0    BMW7812  67.0    Male        208.0         158/88        72.0       0.0   
1    CZE1114  21.0    Male        389.0         165/93        98.0       1.0   
2    BNI9906  21.0  Female        324.0         174/99        72.0       1.0   
3    JLN3497  84.0    Male        383.0        163/100        73.0       1.0   
4    GFO8847  66.0    Male        318.0          91/88        93.0       1.0   
5    ZOO7941  54.0  Female        297.0         172/86        48.0       1.0   
6    WYV0966  90.0    Male        358.0         102/73        84.0       0.0   
7    XXM0972  84.0    Male        220.0         131/68       107.0       0.0   
8    XCQ5937  20.0    Male        145.0        144/105        68.0       1.0   
9    FTJ5456  43.0  Female        248.0         160/70        55.0       0.0   

   Family History  Smoking  Obesity  ...  Sedentary Hours Per Day    Income  \
0             0.0      1.0      0.0  ...                 6.615001  261404.0   
1             1.0      1.0      1.0  ...                 4.963459  285768.0   
2             0.0      0.0      0.0  ...                 9.463426  235282.0   
3             1.0      1.0      0.0  ...                 7.648981  125640.0   
4             1.0      1.0      1.0  ...                 1.514821  160555.0   
5             1.0      1.0      0.0  ...                 7.798752  241339.0   
6             0.0      1.0      0.0  ...                 0.627356  190450.0   
7             0.0      1.0      1.0  ...                10.543780  122093.0   
8             0.0      1.0      1.0  ...                11.348787   25086.0   
9             1.0      1.0      1.0  ...                 4.055115  209703.0   

         BMI  Triglycerides  Physical Activity Days Per Week  \
0  31.251233          286.0                              0.0   
1  27.194973          235.0                              1.0   
2  28.176571          587.0                              4.0   
3  36.464704          378.0                              3.0   
4  21.809144          231.0                              1.0   
5  20.146839          795.0                              5.0   
6  28.885811          284.0                              4.0   
7  22.221862          370.0                              6.0   
8  35.809901          790.0                              7.0   
9  22.558917          232.0                              7.0   

   Sleep Hours Per Day    Country      Continent           Hemisphere  \
0                  6.0  Argentina  South America  Southern Hemisphere   
1                  7.0     Canada  North America  Northern Hemisphere   
2                  4.0     France         Europe  Northern Hemisphere   
3                  4.0     Canada  North America  Northern Hemisphere   
4                  5.0   Thailand           Asia  Northern Hemisphere   
5                 10.0    Germany         Europe  Northern Hemisphere   
6                 10.0     Canada  North America  Northern Hemisphere   
7                  7.0      Japan           Asia  Northern Hemisphere   
8                  4.0     Brazil  South America  Southern Hemisphere   
9                  7.0      Japan           Asia  Northern Hemisphere   

   Heart Attack Risk  
0                0.0  
1                0.0  
2                0.0  
3                0.0  
4                0.0  
5                1.0  
6                1.0  
7                1.0  
8                0.0  
9                0.0  

[10 rows x 26 columns]

# ▶️ Renombrado de columnas al español

In [52]:
# Renombramos las columnas al español
df = df.rename(columns={"Patient ID":"id_paciente",
                        "Age":"edad",
                        "Sex": "sexo",
                        "Cholesterol" : "colesterol",
                        "Blood Pressure": "presion_arterial",
                        "Heart Rate":"frecuencia_cardiaca",
                        "Diabetes":"diabetes",
                        "Family History":"antecedentes_familiares",
                        "Smoking":"tabaquismo",
                        "Obesity":"obesidad",
                        "Alcohol Consumption":"consumo_de_alcohol",
                        "Exercise Hours Per Week":"horas_ejercicio_semanal",
                        "Diet":"dieta",
                        "Previous Heart Problems":"problemas_cardiacos_previos",
                        "Medication Use":"uso_medicación",
                        "Stress Level":"nivel_estres",
                        "Sedentary Hours Per Day":"horas_sedentarismo_diarias",
                        "Income":"ingresos",
                        "BMI":"BMI",
                        "Triglycerides":"trigliceridos",
                        "Physical Activity Days Per Week":"dias_actividad_fisica_semanal",
                        "Sleep Hours Per Day":"horas_sueño_diarias",
                        "Country":"pais",
                        "Continent":"continente",
                        "Hemisphere":"hemisferio",
                        "Heart Attack Risk":"riesgo_infarto"})

## ⏩ Tamaño de filas y columnas del dataset

In [53]:
# Cantidad e columnas
print(f"La cantidad de columnas es de {df.shape[1]}")
# Cantidad de filas
print(f"La cantidad de filas es de {df.shape[0]}")

La cantidad de columnas es de 26
La cantidad de filas es de 8780


## ⏩ Datos estadísticos cuantitativos

In [54]:
# Visualización de los datos estdisticos transpuestos
df.describe().round(2).T

count       mean       std      min       25%  \
edad                           8771.0      53.74     21.43     18.0     35.00   
colesterol                     8772.0     260.24     87.24    120.0    192.00   
frecuencia_cardiaca            8771.0      75.01     20.54     40.0     57.00   
diabetes                       8772.0       0.65      0.48      0.0      0.00   
antecedentes_familiares        8772.0       0.49      0.50      0.0      0.00   
tabaquismo                     8772.0       0.90      0.30      0.0      1.00   
obesidad                       8772.0       0.50      0.50      0.0      0.00   
consumo_de_alcohol             8772.0       0.60      0.49      0.0      0.00   
horas_ejercicio_semanal        8772.0      10.02      5.78      0.0      4.99   
problemas_cardiacos_previos    8772.0       0.50      0.50      0.0      0.00   
uso_medicación                 8772.0       0.50      0.50      0.0      0.00   
nivel_estres                   8772.0       5.47      2.86      1.0      3.00   
horas_sedentarismo_diarias     8772.0       5.99      3.47      0.0      2.98   
ingresos                       8772.0  158276.56  80534.92  20062.0  88355.00   
BMI                            8772.0      28.90      6.32     18.0     23.43   
trigliceridos                  8772.0     428.98   1086.50     30.0    226.00   
dias_actividad_fisica_semanal  8772.0       3.49      2.28      0.0      1.00   
horas_sueño_diarias            8772.0       7.02      1.99      4.0      5.00   
riesgo_infarto                 8772.0       0.36      0.48      0.0      0.00   

                                     50%        75%       max  
edad                               54.00      72.00     182.0  
colesterol                        259.00     330.00    3337.0  
frecuencia_cardiaca                75.00      93.00     110.0  
diabetes                            1.00       1.00       1.0  
antecedentes_familiares             0.00       1.00       1.0  
tabaquismo                          1.00       1.00       1.0  
obesidad                            1.00       1.00       1.0  
consumo_de_alcohol                  1.00       1.00       1.0  
horas_ejercicio_semanal            10.08      15.05      20.0  
problemas_cardiacos_previos         0.00       1.00       1.0  
uso_medicación                      0.00       1.00       1.0  
nivel_estres                        5.00       8.00      10.0  
horas_sedentarismo_diarias          5.92       9.02      12.0  
ingresos                       157978.50  227653.00  299954.0  
BMI                                28.78      34.34      40.0  
trigliceridos                     417.00     612.00   99999.0  
dias_actividad_fisica_semanal       3.00       5.00       7.0  
horas_sueño_diarias                 7.00       9.00      10.0  
riesgo_infarto                      0.00       1.00       1.0

## 📜 Lectura

#### <strong>Heart Rate </strong>La fracuencia cardiaca es un indicador clave de la salud cardiovascular y su variabilidad puede reflejar condiciones físicas o patológicas. En este análisis, el promedio de la frecuencia cardiaca es de 75 bpm, lo cual se encuentra dentro del rango considerado como normal para un adulto en reposo (60 - 100). Sin embargo la desviación estándar de 20.55 bpm indica una variabilidad considerable en los datos. sigiriendo que hay pacientes con frecuencias cardiacas significativamente mas altas o mas baja que la media

## ⏩ Datos estadísticos cualitativos del dataset


In [55]:
# Resumen estadistico de variable cualitativas
df.describe(include="object").T

count unique                  top  freq
id_paciente       8772   8763              ATI9164    10
sexo              8771      3                 Male  6118
presion_arterial  8772   3915                92/71    13
dieta             8771      3              Healthy  2969
pais              8772     20              Germany   477
continente        8772      6                 Asia  2552
hemisferio        8691      2  Northern Hemisphere  5620

### 📜 Lectura


#### <p><strong>Blood Pressure</strong> la cantidad total de registros para Blood Pressure es de 8772, de los cuales 3915 son valores únicos. el valor que mas se repite (moda) en Blood Pressure es 92/71, con una frecuencia de 13 ocurrencias</p>

⏩ Informacion del tipo de dato de cada columna</h2>

In [56]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8780 entries, 0 to 8779
Data columns (total 26 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   id_paciente                    8772 non-null   object 
 1   edad                           8771 non-null   float64
 2   sexo                           8771 non-null   object 
 3   colesterol                     8772 non-null   float64
 4   presion_arterial               8772 non-null   object 
 5   frecuencia_cardiaca            8771 non-null   float64
 6   diabetes                       8772 non-null   float64
 7   antecedentes_familiares        8772 non-null   float64
 8   tabaquismo                     8772 non-null   float64
 9   obesidad                       8772 non-null   float64
 10  consumo_de_alcohol             8772 non-null   float64
 11  horas_ejercicio_semanal        8772 non-null   float64
 12  dieta                          8771 non-null   o

In [57]:
# impresion de el detalle de la cantidad de columas y cada columnas relacionando el tipo de datos de cada columna
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8780 entries, 0 to 8779
Data columns (total 26 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   id_paciente                    8772 non-null   object 
 1   edad                           8771 non-null   float64
 2   sexo                           8771 non-null   object 
 3   colesterol                     8772 non-null   float64
 4   presion_arterial               8772 non-null   object 
 5   frecuencia_cardiaca            8771 non-null   float64
 6   diabetes                       8772 non-null   float64
 7   antecedentes_familiares        8772 non-null   float64
 8   tabaquismo                     8772 non-null   float64
 9   obesidad                       8772 non-null   float64
 10  consumo_de_alcohol             8772 non-null   float64
 11  horas_ejercicio_semanal        8772 non-null   float64
 12  dieta                          8771 non-null   o

## ⏩ Búsqueda de valores duplicados

In [58]:
print(f"La cantidad de valores duplicados encontrados es de {df.duplicated().sum()}")

duplicados = df[df.duplicated()] #capturo los datos duplicados
duplicados # visualizo los datos duplicados

La cantidad de valores duplicados encontrados es de 16


id_paciente  edad  sexo  colesterol presion_arterial  \
100      ATI9164  28.0  Male       276.0            92/71   
101      ATI9164  28.0  Male       276.0            92/71   
102      ATI9164  28.0  Male       276.0            92/71   
103      ATI9164  28.0  Male       276.0            92/71   
104      ATI9164  28.0  Male       276.0            92/71   
105      ATI9164  28.0  Male       276.0            92/71   
106      ATI9164  28.0  Male       276.0            92/71   
107      ATI9164  28.0  Male       276.0            92/71   
109      ATI9164  28.0  Male       276.0            92/71   
167          NaN   NaN   NaN         NaN              NaN   
252          NaN   NaN   NaN         NaN              NaN   
463          NaN   NaN   NaN         NaN              NaN   
746          NaN   NaN   NaN         NaN              NaN   
1375         NaN   NaN   NaN         NaN              NaN   
2633         NaN   NaN   NaN         NaN              NaN   
6135         NaN   NaN   NaN         NaN              NaN   

      frecuencia_cardiaca  diabetes  antecedentes_familiares  tabaquismo  \
100                  65.0       1.0                      0.0         1.0   
101                  65.0       1.0                      0.0         1.0   
102                  65.0       1.0                      0.0         1.0   
103                  65.0       1.0                      0.0         1.0   
104                  65.0       1.0                      0.0         1.0   
105                  65.0       1.0                      0.0         1.0   
106                  65.0       1.0                      0.0         1.0   
107                  65.0       1.0                      0.0         1.0   
109                  65.0       1.0                      0.0         1.0   
167                   NaN       NaN                      NaN         NaN   
252                   NaN       NaN                      NaN         NaN   
463                   NaN       NaN                      NaN         NaN   
746                   NaN       NaN                      NaN         NaN   
1375                  NaN       NaN                      NaN         NaN   
2633                  NaN       NaN                      NaN         NaN   
6135                  NaN       NaN                      NaN         NaN   

      obesidad  ...  horas_sedentarismo_diarias  ingresos        BMI  \
100        1.0  ...                    0.808766  171302.0  34.555965   
101        1.0  ...                    0.808766  171302.0  34.555965   
102        1.0  ...                    0.808766  171302.0  34.555965   
103        1.0  ...                    0.808766  171302.0  34.555965   
104        1.0  ...                    0.808766  171302.0  34.555965   
105        1.0  ...                    0.808766  171302.0  34.555965   
106        1.0  ...                    0.808766  171302.0  34.555965   
107        1.0  ...                    0.808766  171302.0  34.555965   
109        1.0  ...                    0.808766  171302.0  34.555965   
167        NaN  ...                         NaN       NaN        NaN   
252        NaN  ...                         NaN       NaN        NaN   
463        NaN  ...                         NaN       NaN        NaN   
746        NaN  ...                         NaN       NaN        NaN   
1375       NaN  ...                         NaN       NaN        NaN   
2633       NaN  ...                         NaN       NaN        NaN   
6135       NaN  ...                         NaN       NaN        NaN   

      trigliceridos  dias_actividad_fisica_semanal  horas_sueño_diarias  \
100           385.0                            1.0                  5.0   
101           385.0                            1.0                  5.0   
102           385.0                            1.0                  5.0   
103           385.0                            1.0                  5.0   
104           385.0                            1.0                  5.0   

### ⏩ Eliminación de valores duplicados
### Se deciden eliminar los valores duplicados ya que se detecta que los mismo son registros que duplicados por pertenecer al mismo ID de paciente y detectar que el resto de los registros son valores en blanco

In [59]:
df = df.drop_duplicates() # elimino duplicados y sobreescribo el dataframe
df.duplicated().sum() # Verifico nuevamente la suma de los duplicados
print(f"La cantidad de valores duplicados encontrados es de {df.duplicated().sum()}")

La cantidad de valores duplicados encontrados es de 0


## ⏩ Búsqueda de valores nulos

In [60]:
# imprimimos la suma total de los valores nulos encontrados
print(f"La cantidad de valores nulos encontrados es de {df.isnull().sum().sum()}")

df.isnull().sum()


La cantidad de valores nulos encontrados es de 111


id_paciente                       1
edad                              2
sexo                              2
colesterol                        1
presion_arterial                  1
frecuencia_cardiaca               2
diabetes                          1
antecedentes_familiares           1
tabaquismo                        1
obesidad                          1
consumo_de_alcohol                1
horas_ejercicio_semanal           1
dieta                             2
problemas_cardiacos_previos       1
uso_medicación                    1
nivel_estres                      1
horas_sedentarismo_diarias        1
ingresos                          1
BMI                               1
trigliceridos                     1
dias_actividad_fisica_semanal     1
horas_sueño_diarias               1
pais                              1
continente                        1
hemisferio                       82
riesgo_infarto                    1
dtype: int64

In [61]:
# Buscamos y sumamos los valores nulos en caso de existir
nulos = df[df.isnull().any(axis=1)] # Busco y guardo la fila nula
nulos


id_paciente  edad    sexo  colesterol presion_arterial  \
27       JDP9221  40.0     NaN       273.0           160/76   
108          NaN   NaN     NaN         NaN              NaN   
3763     XGJ2522  59.0    Male       162.0           145/98   
3764     OJX0002  27.0    Male       275.0           156/89   
3765     WPN0485  64.0    Male       246.0           165/65   
...          ...   ...     ...         ...              ...   
4559     KSL0012  58.0  Female       372.0           152/72   
4560     ODP5325  46.0    Male       362.0           177/63   
4561     HDI0288  20.0    Male       170.0           150/70   
4562     TSP5679  85.0    Male       187.0           114/89   
4563     WCY3053  79.0    Male       354.0           160/70   

      frecuencia_cardiaca  diabetes  antecedentes_familiares  tabaquismo  \
27                   96.0       0.0                      0.0         1.0   
108                   NaN       NaN                      NaN         NaN   
3763                 95.0       1.0                      1.0         1.0   
3764                 81.0       0.0                      0.0         1.0   
3765                 55.0       1.0                      0.0         1.0   
...                   ...       ...                      ...         ...   
4559                 95.0       0.0                      1.0         1.0   
4560                 87.0       1.0                      0.0         1.0   
4561                 64.0       1.0                      1.0         1.0   
4562                 69.0       1.0                      1.0         1.0   
4563                 51.0       1.0                      1.0         1.0   

      obesidad  ...  horas_sedentarismo_diarias  ingresos        BMI  \
27         1.0  ...                    4.547242  178241.0  26.417047   
108        NaN  ...                         NaN       NaN        NaN   
3763       1.0  ...                    1.351678  102136.0  18.944005   
3764       1.0  ...                    3.992922  146579.0  30.720929   
3765       0.0  ...                    4.680722   62849.0  35.974714   
...        ...  ...                         ...       ...        ...   
4559       0.0  ...                    7.762439  272411.0  30.240878   
4560       1.0  ...                    2.823926  180080.0  19.939337   
4561       1.0  ...                    2.970027  171058.0  29.004499   
4562       0.0  ...                    2.639639   86795.0  34.590190   
4563       0.0  ...                   10.461988   80402.0  22.597108   

      trigliceridos  dias_actividad_fisica_semanal  horas_sueño_diarias  \
27            551.0                            4.0                  6.0   
108             NaN                            NaN                  NaN   
3763          611.0                            3.0                  7.0   
3764          767.0                            1.0                  7.0   
3765          543.0                            5.0                  4.0   
...             ...                            ...                  ...   
4559          209.0                            4.0                  8.0   
4560          471.0                            7.0                  5.0   
4561          797.0                            7.0                 10.0   
4562           67.0                            4.0                  9.0   
4563          424.0                            1.0                  7.0   

              pais     continente           hemisferio  riesgo_infarto  
27          France         Europe  Northern Hemisphere             0.0  
108            NaN            NaN                  NaN             NaN  
3763         Italy         Europe                  NaN             0.0  
3764     Argentina  South America                  NaN             1.0  
3765         Spain         Europe                  NaN             0.0  
...            ...            ...                  ...             ...  
4559     Argentina  South America                  NaN    

### ⏩ Eliminación de valores nulos

In [62]:
# Eliminación de fila completa por valores nulos
df = df.drop(index=108, axis=0)
# borrado de columna completa Hemisferio ya que es irrelevante en el análisis
df = df.drop(columns="hemisferio", axis=1)
# paso anterior para buscar nuevamente los resultados faltantes para posterior tratamiento
nulos = df[df.isnull().any(axis=1)] # Busco y guardo la fila nula
nulos

id_paciente  edad  sexo  colesterol presion_arterial  \
27       JDP9221  40.0   NaN       273.0           160/76   
4554     OXY2061   NaN  Male       133.0           156/82   

      frecuencia_cardiaca  diabetes  antecedentes_familiares  tabaquismo  \
27                   96.0       0.0                      0.0         1.0   
4554                  NaN       1.0                      1.0         1.0   

      obesidad  ...  nivel_estres  horas_sedentarismo_diarias  ingresos  \
27         1.0  ...           4.0                    4.547242  178241.0   
4554       0.0  ...           7.0                    9.025707  134456.0   

            BMI  trigliceridos  dias_actividad_fisica_semanal  \
27    26.417047          551.0                            4.0   
4554  35.475337          645.0                            6.0   

      horas_sueño_diarias    pais     continente  riesgo_infarto  
27                    6.0  France         Europe             0.0  
4554                  7.0  Brazil  South America             0.0  

[2 rows x 25 columns]

## ⏩ Corrección de valores faltantes